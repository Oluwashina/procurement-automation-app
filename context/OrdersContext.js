import React,{createContext, useState, useCallback, useEffect, useContext} from 'react'


const OrdersContext = createContext()

export const useOrders = () =>{
    return useContext(OrdersContext)
}

export const OrdersProvider = ({children}) =>{

    const [orders, setOrders] = useState([])
    const [loading, setLoading] = useState(false);

    const fetchOrders = useCallback(async () => {
        try {
            const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/orders`);
            if (response.ok) {
                const data = await response.json();
                setOrders(data);
                setLoading(false);
            } else {
                console.log('Failed to fetch orders.');
                setLoading(false);
            }
            } catch (error) {
                console.log('Failed to fetch orders.');
            setLoading(false);
         }
      }, []);
    
      useEffect(() => {
        fetchOrders();
      }, [fetchOrders]);

    return(
        <OrdersContext.Provider value={{orders, loading, fetchOrders}}>
            {children}
        </OrdersContext.Provider>
    )
}