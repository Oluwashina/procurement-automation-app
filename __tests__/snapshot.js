import { render } from '@testing-library/react'
import Home from '../pages/index'
import { OrdersProvider } from '@/context/OrdersContext'

 
it('renders homepage unchanged', () => {
  const { container } = render(
        <OrdersProvider>
            <Home />
        </OrdersProvider>
 )
  expect(container).toMatchSnapshot()
})