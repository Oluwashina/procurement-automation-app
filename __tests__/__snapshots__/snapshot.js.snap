// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="className"
  >
    <div
      class="container"
    >
      <div>
        <h6
          class="title"
        >
          Orders
        </h6>
        <p
          class="subtitle"
        >
          Showing data over the last 30 days
        </p>
      </div>
      <div
        class="container_cta"
      >
        <div>
          <button
            class="primary_btn"
          >
            <img
              alt="Add Icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
              style="color: transparent;"
              width="40"
            />
            New Order
          </button>
        </div>
        <div>
          <button
            class="secondary_btn"
          >
            <img
              alt="Compass Icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
              style="color: transparent;"
              width="40"
            />
            Track Order
          </button>
        </div>
      </div>
    </div>
    <div
      class="cardLayout"
    >
      <div
        class="card"
      >
        <div>
          <h6>
            Total Orders
          </h6>
          <h4>
            93,000
          </h4>
          <div
            class="cardBadgeDiv"
          >
            <div
              class="cardBagde"
            >
              <img
                alt="Arrow Up Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
                style="color: transparent;"
                width="40"
              />
              5%
            </div>
            <p>
              high today
            </p>
          </div>
        </div>
        <div
          class="cardIcon"
        >
          <img
            alt="Box Icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
      <div
        class="card"
      >
        <div>
          <h6>
            Orders Fulfilled
          </h6>
          <h4>
            80,000
          </h4>
          <div
            class="cardBadgeDiv"
          >
            <div
              class="cardBagde"
            >
              <img
                alt="Arrow Up Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
                style="color: transparent;"
                width="40"
              />
              5%
            </div>
            <p>
              high today
            </p>
          </div>
        </div>
        <div
          class="cardIcon"
        >
          <img
            alt="bag Icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
      <div
        class="card"
      >
        <div>
          <h6>
            Orders In Transit
          </h6>
          <h4>
            30,000
          </h4>
          <div
            class="cardBadgeDiv"
          >
            <div
              class="cardBagde"
            >
              <img
                alt="Arrow Up Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
                style="color: transparent;"
                width="40"
              />
              5%
            </div>
            <p>
              high today
            </p>
          </div>
        </div>
        <div
          class="cardIcon"
        >
          <img
            alt="Lorry Icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=100"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=100 1x, /_next/image?url=%2Fimg.jpg&w=96&q=100 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
    </div>
    <div
      class="ordersDiv"
    >
      <h6
        class="ordersTitle"
      >
        All Orders
      </h6>
      <div
        class="orderTable"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th
                class="tablehead"
                scope="col"
              >
                Order ID/ Customer
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Item Name
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Item Qty
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Date Added
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Payment Status
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Delivery Status
              </th>
              <th
                class="tablehead"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="tablebody"
          />
        </table>
      </div>
    </div>
  </div>
</div>
`;
